playwright/_impl/_json_pipe.py

```diff
@@ -48,18 +46,15 @@ async def wait_until_stopped(self) -> None:
     async def connect(self) -> None:
         self._stopped_future: asyncio.Future = asyncio.Future()
 
-        close_error: Optional[str] = None
+        def handle_closed(close_error: str = None) -> None:
+            self.emit("close", close_error)
```

Comment:
I think you are emitting multiple close events now.

Reply:
Verified that we only emit it a single time. I can protect it if you want that we do it only a single time for safety.